src/Parser/Hunter/Marksmanship/Modules/Traits/Bullseye.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import StatisticBox from "Main/StatisticBox";
+import SpellIcon from "common/SpellIcon";
+import { formatNumber, formatPercentage } from "common/format";
+import SpellLink from 'common/SpellLink';
+import CooldownTracker from '../Features/CooldownThroughputTracker';
+
+class Bullseye extends Analyzer{
+  executeTimestamp;
+  bullseyeResets = 0; //only resets when boss < 20% health, so resets we can confirm shouldn't have happened
+  bullseyeInstances = [];
+  saveForExecuteThreshhold;
+  bossIDs = [];
+  MAX_STACKS = 30;
+  EXECUTE_PERCENT = .2;
+  CROWS_SAVE_PERCENT = .25;
+  static dependencies = {
+    cooldownTracker : CooldownTracker,
+  };
+
+  on_initialized(){
+    this.owner.report.enemies.forEach(enemy => {
+      enemy.fights.forEach(fight => {
+        if (fight.id === this.owner.fight.id && enemy.type === "Boss") this.bossIDs.push(enemy.id);
```

Comment:
It's preferred to always use brackets around if-statements (and multiline them).